index.d.ts

```diff
@@ -0,0 +1,124 @@
+import {JsonValue} from 'type-fest';
+import {BrowserWindow, BrowserView, IpcMain, IpcRenderer} from 'electron';
+
+declare namespace ipc {
+	interface MainProcessIpc extends IpcMain {
+		/**
+		Send a message to the given window.
+
+		In the renderer process, use `ipc.answerMain` to reply to this message.
+
+		@param window - The window to send the message to.
+		@param channel - The channel to send the message on.
+		@param data - Data to send to the receiver.
+
+		@example
+		```
+		import {BrowserWindow} from 'electron';
+		import * as betterIpc from 'electron-better-ipc';
+
+		const ipc: betterIpc.MainProcessIpc = betterIpc;
+		const window = BrowserWindow.getFocusedWindow();
+
+		(async () => {
+			const emoji = await ipc.callRenderer(window!, 'get-emoji', 'unicorn');
+			console.log(emoji);
+			//=> '🦄'
+		})();
+		```
+		*/
+		callRenderer(
+			window: BrowserWindow,
+			channel: string,
+			data?: JsonValue
```

Comment:
I don't think we should limit it to `JsonValue` here. It becomes annoying when you want to just rely on `JSON.stringify`'s serialization. I think `unknown` would be better.

Reply:
What's the problem with relying on JSON serialization for the `data` property? The data is actually serialized to JSON so this is accurate.

I think that we should leave the type of `data` parameter. We could, however change the return type to `Promise<unknown>` following the same reasoning as [here](https://github.com/sindresorhus/electron-better-ipc/pull/13#discussion_r275363493).